### ISSUE TITLE
markdown break some rules

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#3365]
**Describe the bug? (A clear and concise description of what the bug is)**

Some rules buddy with non css `postcss` syntax plugins:

- postcss-jsx
- postcss-html
- postcss-markdown

**Which rule, if any, is this issue related to?**

All rules based on `root.source.input.css` property:
- max-line-length
- no-empty-source
- no-missing-end-of-source-newline

Also looks we break `fix` (need add test)

**What CSS is needed to reproduce this issue?**

Create `test.md` file with content:
---
# title

```css
::placeholder {
	color: gray;
}
```

Very very very very very very very very very very very very very very  very very very very very very very very very very very long line

---

**What stylelint configuration is needed to reproduce this issue?**

e.g.

```json
{
  "rules": {
    "max-line-length": [120, { "ignore": ["comments"] }]
  }
}
```

**Which version of stylelint are you using?**

e.g. `9.2.1`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

not matter

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

no

**What did you expect to happen?**

No warnings to be flagged.

**What actually happened (e.g. what warnings or errors you are getting)?**

```shell
test.md
 3:135  ✖  Expected line length to be no more than 120 characters   max-line-length
```

**Comment by evilebottnawi** [comment]
Looks `root.source.input.css` should contain only css, original content should be stored in `root.source.input.original`

**Comment by evilebottnawi** [comment]
/cc @ai it is normal when `root` node contains in `nodes` other `root` nodes?
Example of markdown file:
---
# title

```css
::placeholder {
	color: gray;
}
```

```css
::placeholder {
	color: gray;
}
```

Very very very very very very very very very very very very very very  very very very very very very very very very very very long line
---

We return one `root` which contain two `root` nodes (each present `css` code block)

**Comment by ai** [comment]
@evilebottnawi not sure. I didn’t plan it in AST. But all walk methods will work and I didn’t think it is a big problem.

**Comment by evilebottnawi** [comment]
@ai i.e. we can rely on this behavior?

**Comment by ai** [comment]
I think so. maybe in the future we will find some problem and will add extra node type for this cases. But right now Root with Roots seems working

**Comment by evilebottnawi** [comment]
Sorry I was a little inconsiderate, need fix for:
- max-line-length
- no-missing-end-of-source-newline

WIP

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/3367

**Comment by gucong3000** [comment]
> it is normal when root node contains in nodes other root nodes?
> we can rely on this behavior?

This is what `postcss-syntax` does for non-css synxtaxes that can use `css` as a part.
